ui/store/actions.ts

```diff
@@ -4851,9 +4852,41 @@ export function getGasFeeTimeEstimate(
   ]);
 }
 
-export async function closeNotificationPopup() {
+export async function attemptCloseNotificationPopup() {
+  // Check if the current window is NOT a popup - if confirmed, we should not close it
+  try {
+    const currentWindow = await browser.windows.getCurrent();
+    if (currentWindow.type && currentWindow.type !== 'popup') {
+      console.warn(
+        `Not safe to close a window that is not a popup: It is of type: ${currentWindow.type}`,
+      );
+      // We've confirmed this is not a popup window, so it's not safe to close.
+      return;
+    }
+  } catch (error) {
+    // Error occurred while checking window type - we cannot confirm it's not a popup, so continue
+    // with the closing attempt as we haven't ruled out that it might be a popup
+  }
```

Comment:
@davidmurdoch, I made the changes we spoke about. But in addition, added this check:

// Check if the current window is NOT a popup - if confirmed, we should not close it.

... since there's no point in closing a window that is not a popup.